lib/utils/sortingUtils.js

```diff
@@ -0,0 +1,103 @@
+
+export function sortByType(array, type) {
+
+  // A custom sort function that sorts elements with $type equal to "type" to the front
+  return array.sort((a, b) => {
+    if (a.$type == type && b.$type != type) {
+      return -1;
+    }
+    if (a.$type != type && b.$type == type) {
+      return 1;
+    }
+    return 0;
+  });
+}
+
+export function sortByPosition(grid, array) {
+  return array.sort((a, b) => {
+    const aPosition = grid.find(a);
+    const bPosition = grid.find(b);
+    if (aPosition[1] !== bPosition[1]) {
```

Comment:
This does not return "0", is this intended? 

It would be benefitial to explain what this function does, and name the argument in a semantic manner.

Reply:
This not returning 0 should be fine, because the positions should be different to begin with.
First, the columns are compared. If the columns are different, the column index is used for sorting. If the columns are the same, the row index is used for sorting.
If both the column and row index are the same, the positions would be identical, which means something went wrong elsewhere. But a grid node can only have one element, so this should not happen.

Reply:
done

I added the logic to return 0, but that should not happen anyway